docs/4.2. Behaviour - Querying.md

```diff
@@ -2,63 +2,63 @@
+Connection upgrade procedures permitting an existing HTTP query to transition to a WebSocket are not currently supported, but could be in the future.
 
 ### Query API Behaviour
-Long-lived connections to the Query API are supported via Websockets which can be set up via the /subscriptions resource. Query APIs may additionally support the HTTP 'Upgrade' header sent by clients to upgrade an HTTP GET request to a Websocket. In cases where this is performed, a corresponding entry in /subscriptions must also be created with matching query parameters.
 
-Upon receiving a request for a new subscription, the Query API should identify whether it already has any websockets open which provide for the requested query. If a websocket exists, the 'subscription' object representing it may be returned to the user. If a relevant websocket does not exist a new one should be created.
+Long-lived connections to the Query API are supported via WebSockets which can be set up via the /subscriptions resource. Query APIs MAY additionally support the HTTP 'Upgrade' header sent by clients to upgrade an HTTP GET request to a WebSocket. In cases where this is performed, a corresponding entry in /subscriptions MUST also be created with matching query parameters.
 
-In normal operation, once a websocket has no more clients subscribed to it the Query API may automatically remove it and its corresponding HTTP-advertised subscription object. The Query API MUST NOT acknowledge HTTP DELETE requests for websockets running in this 'non-persistent' mode, instead issuing an HTTP 403 response.
+Upon receiving a request for a new subscription, the Query API SHOULD identify whether it already has any WebSockets open which provide for the requested query. If a WebSocket exists, the 'subscription' object representing it MAY be returned to the user. If a relevant WebSocket does not exist a new one SHOULD be created.
 
-If a 'persistent' websocket has been requested by the client, this must not be cleaned up automatically by the Query API, even if all consuming clients have disconnected. The user may request closing of this socket by issuing an HTTP DELETE. If an HTTP DELETE is issued prior to all websocket connections being closed, they should be forcibly closed by the server.
+In normal operation, once a WebSocket has no more clients subscribed to it the Query API MAY automatically remove it and its corresponding HTTP-advertised subscription object. The Query API MUST NOT acknowledge HTTP DELETE requests for WebSockets running in this 'non-persistent' mode, instead issuing an HTTP 403 response.
 
-### Websocket Protocol
+If a 'persistent' WebSocket has been requested by the client, this MUST NOT be cleaned up automatically by the Query API, even if all consuming clients have disconnected. The user MAY request closing of this socket by issuing an HTTP DELETE. If an HTTP DELETE is issued prior to all WebSocket connections being closed, they SHOULD be forcibly closed by the server.
```

Comment:
I have the same slight confusion here about what "a WebSocket" is. I'd rewrite those paragraphs with WebSocket replaced by subscription (and e.g. change 'running' to 'configured').

Reply:
Will look at these.

Reply:
Added suggested versions for two paras, but I'm not sure about the third one. I think this comes down to what 'persistent' can apply to. @andrewbonney any comments?

Reply:
Would be better resolved through #162